### PR TITLE
feat(build-and-deploy-container-image): start NATS for testing

### PR DESCRIPTION
**Pull request by RappC**
Tested here: https://github.com/neohelden/NEAP/pull/369

### CODE REVIEW

**Review by TheDome — CHANGES_REQUESTED** [review]

File: build-and-test-with-yarn/action.yaml [hunk 1]
```diff
@@ -6,6 +6,10 @@ inputs:
     description: Ref to checkout from Git
     default: ${{ github.event.pull_request.head.sha }}
     required: false
+  start-nats:
+    description: Set to `true` if you need NATS started for tests
+    default: "false"
+    required: false
```
[the remark this answers is not in the document]
Comment (RappC) [reply build-and-test-with-yarn/action.yaml]: deleted

File: .github/workflows/build-and-deploy-container-image.yaml [hunk 2]
```diff
@@ -224,8 +224,29 @@ jobs:
...
+        credentials:
+          username:  ${{ secrets.image-registry-username }}
+          password: ${{ secrets.image-registry-token }}
+        ports:
+          - 8222:8222
+          - 4222:4222
+    env:
+      # Dont judge me https://github.com/nats-io/nats-docker/blob/4fe4d656443e2a43caab61ecd665436e12e3358a/2.2.6/alpine3.13/nats-server.conf
+      NPQ_PASS: T0pS3cr3t
+      NPQ_USER: ruser
```
[the remark this answers is not in the document]
Comment (RappC) [reply .github/workflows/build-and-deploy-container-image.yaml]: updated

File: .github/workflows/build-and-deploy-container-image.yaml [hunk 3]
```diff
@@ -224,7 +224,28 @@ jobs:
...
+      npq:
+        image: registry.neohelden.com/neohelden/nats:2.6.3-2.0.0
+        credentials:
+          username:  ${{ secrets.image-registry-username }}
+          password: ${{ secrets.image-registry-token }}
+        ports:
+          - 8222:8222
+          - 4222:4222
+    env:
+      # Dont judge me https://github.com/nats-io/nats-docker/blob/4fe4d656443e2a43caab61ecd665436e12e3358a/2.2.6/alpine3.13/nats-server.conf
```
Comment (TheDome) [inline .github/workflows/build-and-deploy-container-image.yaml:237]: We should change this
```suggestion
      # Dont judge me https://github.com/neohelden/nats/blob/eee9e594d6b451c43192ba05f5cb0da609a8b38d/Dockerfile#L5
```

**Review by TheDome — CHANGES_REQUESTED** [review]
LGTM

**Review by TheDome — APPROVED** [review]

**Pull request merged**